CI: Runtime Test (privileged) - Test_NeighborGracefulRestart - `did not receive expected peering state ["ESTABLISHED"]`

## CI failure

```
--- FAIL: Test_NeighborGracefulRestart (60.05s)
    --- FAIL: Test_NeighborGracefulRestart/add_neighbor_with_defaults (59.49s)
        neighbor_test.go:309: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:309
            	            				/root/sdk/go1.21.1/src/runtime/asm_amd64.s:1650
            	Error:      	"[]" should have 1 item(s), but has 0
            	Test:       	Test_NeighborGracefulRestart/add_neighbor_with_defaults
        neighbor_test.go:325: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:325
            	Error:      	Condition never satisfied
            	Test:       	Test_NeighborGracefulRestart/add_neighbor_with_defaults
    --- FAIL: Test_NeighborGracefulRestart/update_graceful_restart_with_defaults (0.00s)
        neighbor_test.go:298: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:298
            	Error:      	Received unexpected error:
            	            	did not receive expected peering state ["ESTABLISHED"], context deadline exceeded
            	Test:       	Test_NeighborGracefulRestart/update_graceful_restart_with_defaults
    --- FAIL: Test_NeighborGracefulRestart/update_graceful_restart,_restart_time (0.00s)
        neighbor_test.go:298: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:298
            	Error:      	Received unexpected error:
            	            	did not receive expected peering state ["ESTABLISHED"], context deadline exceeded
            	Test:       	Test_NeighborGracefulRestart/update_graceful_restart,_restart_time
    --- FAIL: Test_NeighborGracefulRestart/disable_graceful_restart (0.00s)
        neighbor_test.go:298: 
            	Error Trace:	/host/pkg/bgpv1/test/neighbor_test.go:298
            	Error:      	Received unexpected error:
            	            	did not receive expected peering state ["ESTABLISHED"], context deadline exceeded
            	Test:       	Test_NeighborGracefulRestart/disable_graceful_restart
FAIL
```

Observed in a PR which does not contain any code changes: https://github.com/cilium/cilium/actions/runs/6418224530/job/17425610778

cc @cilium/sig-bgp

--- a/pkg/bgpv1/manager/manager.go
+++ b/pkg/bgpv1/manager/manager.go
@@ -79,6 +79,9 @@ type BGPRouterManager struct {
 	lock.RWMutex
 	Servers     LocalASNMap
 	Reconcilers []reconciler.ConfigReconciler
+
+	// running is set when the manager is running, and unset when it is stopped.
+	running bool
 }
 
 // NewBGPRouterManager constructs a GoBGP-backed BGPRouterManager.
@@ -118,6 +121,7 @@ func NewBGPRouterManager(params bgpRouterManagerParams) agent.BGPRouterManager {
 	return &BGPRouterManager{
 		Servers:     make(LocalASNMap),
 		Reconcilers: activeReconcilers,
+		running:     true, // start with running state set
 	}
 }
 
@@ -135,6 +139,10 @@ func (m *BGPRouterManager) ConfigurePeers(ctx context.Context,
 	ciliumNode *v2api.CiliumNode) error {
 	m.Lock()
 	defer m.Unlock()
+
+	if !m.running {
+		return fmt.Errorf("bgp router manager is not running")
+	}
 
 	l := log.WithFields(
 		logrus.Fields{
@@ -392,6 +400,10 @@ func (m *BGPRouterManager) GetPeers(ctx context.Context) ([]*models.BgpPeer, err
 	m.RLock()
 	defer m.RUnlock()
 
+	if !m.running {
+		return nil, fmt.Errorf("bgp router manager is not running")
+	}
+
 	var res []*models.BgpPeer
 
 	for _, s := range m.Servers {
@@ -408,6 +420,10 @@ func (m *BGPRouterManager) GetPeers(ctx context.Context) ([]*models.BgpPeer, err
 func (m *BGPRouterManager) GetRoutes(ctx context.Context, params restapi.GetBgpRoutesParams) ([]*models.BgpRoute, error) {
 	m.RLock()
 	defer m.RUnlock()
+
+	if !m.running {
+		return nil, fmt.Errorf("bgp router manager is not running")
+	}
 
 	// validate router ASN
 	if params.RouterAsn != nil {
@@ -452,6 +468,10 @@ func (m *BGPRouterManager) GetRoutePolicies(ctx context.Context, params restapi.
 	m.RLock()
 	defer m.RUnlock()
 
+	if !m.running {
+		return nil, fmt.Errorf("bgp router manager is not running")
+	}
+
 	// validate router ASN
 	if params.RouterAsn != nil {
 		if _, found := m.Servers[*params.RouterAsn]; !found {
@@ -483,4 +503,5 @@ func (m *BGPRouterManager) Stop() {
 	}
 
 	m.Servers = make(LocalASNMap)
+	m.running = false
 }

--- a/pkg/bgpv1/manager/manager_test.go
+++ b/pkg/bgpv1/manager/manager_test.go
@@ -171,6 +171,7 @@ func TestGetRoutes(t *testing.T) {
 				Servers: map[int64]*instance.ServerWithConfig{
 					int64(testRouterASN): testSC,
 				},
+				running: true,
 			}
 
 			// add a neighbor

